src/server.js

```diff
@@ -88,14 +183,203 @@ export class Server {
+                      .assetBought()
+                      .value()
+                      .assetCode()
+                      .toString()
+                      .replace(/\0/g, '');
+                    assetBought.issuer = StrKey.encodeEd25519PublicKey(
+                      offerClaimed
+                        .assetBought()
+                        .value()
+                        .issuer()
+                        .ed25519()
+                    );
+                  }
+
+                  return {
+                    sellerId: StrKey.encodeEd25519PublicKey(
+                      offerClaimed.sellerId().ed25519()
+                    ),
+                    offerId: offerClaimed.offerId().toString(),
+                    assetSold,
+                    amountSold: _getAmountInLumens(claimedOfferAmountSold),
+                    assetBought,
+                    amountBought: _getAmountInLumens(claimedOfferAmountBought)
+                  };
+                });
+
+              const effect = offerSuccess.offer().switch().name;
+
+              let currentOffer;
+
+              if (
+                typeof offerSuccess.offer().value === 'function' &&
+                offerSuccess.offer().value()
+              ) {
+                const offerXDR = offerSuccess.offer().value();
+
+                currentOffer = {
+                  offerId: offerXDR.offerId().toString(),
+                  selling: {
+                    type: offerXDR.selling().switch().name
+                  },
+                  buying: {
+                    type: offerXDR.buying().switch().name
+                  },
+                  amount: _getAmountInLumens(offerXDR.amount().toString()),
+                  price: {
+                    n: offerXDR.price().n(),
+                    d: offerXDR.price().d()
+                  }
+                };
+
+                if (currentOffer.selling.type !== 'assetTypeNative') {
+                  currentOffer.selling.assetCode = offerXDR
+                    .selling()
+                    .value()
+                    .assetCode()
+                    .toString()
+                    .replace(/\0/g, '');
+                  currentOffer.selling.issuer = StrKey.encodeEd25519PublicKey(
+                    offerXDR
+                      .selling()
+                      .value()
+                      .issuer()
+                      .ed25519()
+                  );
+                }
+
+                if (currentOffer.buying.type !== 'assetTypeNative') {
+                  currentOffer.buying.assetCode = offerXDR
+                    .buying()
+                    .value()
+                    .assetCode()
+                    .toString()
+                    .replace(/\0/g, '');
+                  currentOffer.buying.issuer = StrKey.encodeEd25519PublicKey(
+                    offerXDR
+                      .buying()
+                      .value()
+                      .issuer()
+                      .ed25519()
```

Comment:
There are many repetitions of this code. Maybe move it to a function that accepts XDR asset and returns `{type, assetCode, issuer}` object.

Reply:
Actually, `Asset.fromOperation` in base is something you could use here.

Reply:
Agreed, I had the same comment.

Reply:
Wow, Asset.fromOperation was much better, thanks!